File: payroll.R
# Generate 400 workers dynamically
generate_workers <- function(num_workers = 400) {
  workers <- data.frame(
    id = 1:num_workers,
    name = paste("Worker", 1:num_workers, sep = "_"),
    gender = sample(c("Male", "Female"), num_workers, replace = TRUE),
    salary = sample(5000:35000, num_workers, replace = TRUE),
    stringsAsFactors = FALSE
  )
  return(workers)
}

# Function to assign employee level
assign_level <- function(salary, gender) {
  if (salary > 10000 & salary < 20000) {
    return("A1")
  } else if (salary > 7500 & salary < 30000 & gender == "Female") {
    return("A5-F")
  } else {
    return("General")
  }
}

# Generate payment slips
generate_payment_slips <- function(workers) {
  for (i in 1:nrow(workers)) {
    worker <- workers[i, ]
    level <- tryCatch({
      assign_level(worker$salary, worker$gender)
    }, error = function(e) {
      "Unknown"
    })
    
    cat("Payment Slip for", worker$name, "\n")
    cat("ID:", worker$id, "\n")
    cat("Gender:", worker$gender, "\n")
    cat("Salary: $", worker$salary, "\n", sep = "")
    cat("Level:", level, "\n")
    cat("----------------------------------------\n")
  }
}

# Main Program
workers_list <- generate_workers(400)
generate_payment_slips(workers_list)

File: payroll.py
import random

# Step 1 & 2: Create a list of at least 400 workers dynamically
# Each worker has a name, gender, and salary
workers = []
genders = ["Male", "Female"]

for i in range(1, 401):
    worker = {
        "id": i,
        "name": f"Worker_{i}",
        "gender": random.choice(genders),
        "salary": random.randint(5000, 35000)  # random salary between 5k and 35k
    
    }
    workers.append(worker)

# Step 3: Generate payment slips for all workers
for worker in workers:
    try:
        salary = worker["salary"]
        gender = worker["gender"]
        emp_level = "Unassigned"

        # Step 4: Apply conditions
        if 10000 < salary < 20000:
            emp_level = "A1"
        if 7500 < salary < 30000 and gender == "Female":
            emp_level = "A5-F"

        # Payment slip
        print(f"Payment Slip - {worker['name']}")
        print(f"ID: {worker['id']}")
        print(f"Gender: {gender}")
        print(f"Salary: ${salary}")
        print(f"Employee Level: {emp_level}")
        print("-" * 40)

    except KeyError as e:
        print(f"Error: Missing key {e} in worker data for worker {worker.get('id', 'Unknown')}")
    except Exception as e:
        print(f"Unexpected error for worker {worker.get('id', 'Unknown')}: {e}")